addon/usables/use-machine.ts

```diff
@@ -150,26 +191,15 @@ export class Statechart<
           return cancel.call(null, timer);
         },
       },
-    }).onTransition((state) => {
-      this.state = state;
     });
 
-    this.onTransition(this[CONFIG]?.onTransition);
-
-    this[INTERPRETER].start(this[CONFIG]?.initialState);
-  }
-
-  /**
-   * Lifecycle methods on Resource
-   *
-   */
-  @action
-  protected setup() {
-    let statechart = this.args.positional?.[0] || this.args.named?.chart;
+    this[INTERPRETER].onTransition((state) => {
+      this.state = state;
+    });
 
-    assert(ERROR_CHART_MISSING, statechart);
+    this.onTransition(config?.onTransition);
 
-    this[MACHINE] = createMachine(statechart);
+    this[INTERPRETER].start();
```

Comment:
I don't think interpreter should be started in setup it won't be compat with when we can do withContext again

Reply:
@NullVoxPopuli that's correct -- if we want to keep using `withContext` this wouldn't work for that. I'm not tied to continuing to use `withContext`, so I removed it, but if keeping that builder style API is a goal, you're right -- this won't work.